app/main.py

```diff
@@ -1,3 +1,68 @@
+from __future__ import annotations
+from typing import Any
+
+
 class Distance:
-    # Write your code here
-    pass
+    def __init__(
+            self,
+            km: int | float
+    ) -> None:
+        self.km = km
+
+    @staticmethod
+    def other_type_check(other: Any[Distance, float, int]) -> bool:
+        if not isinstance(other, (Distance, float, int)):
+            raise TypeError("'other' is not 'Distance', 'float', 'int'!!!")
+        return isinstance(other, Distance)
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other: Any[Distance, float, int]) -> Distance:
+        return Distance(self.km + other.km) \
```

Comment:
Avoid using backslashes for line continuation. It is recommended to use parentheses for line continuation. This applies to all the places in your code where you've used backslashes.